test/accept-from.spec.js

```diff
@@ -0,0 +1,85 @@
+const {expect} = require('chai')
+const sinon = require('sinon')
+const Gossipsub = require('../src')
+
+describe('Gossipsub acceptFrom', () => {
+  let gossipsub
+  let sandbox
+  let scoreSpy
+
+  beforeEach(async () => {
+    sandbox = sinon.createSandbox()
+    sandbox.useFakeTimers(Date.now())
+    gossipsub = new Gossipsub({}, { emitSelf: false })
+    // stubbing PeerScore causes some pending issue in firefox browser environment
+    // we can only spy it
+    // using scoreSpy.withArgs("peerA").calledOnce causes the pending issue in firefox
+    // while spy.getCall() is fine
+    scoreSpy = sandbox.spy(gossipsub.score, "score")
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it('should only white list peer with positive score', () => {
+    // by default the score is 0
+    gossipsub._acceptFrom("peerA")
+    // 1st time, we have to compute score
+    expect(scoreSpy.getCall(0).args[0]).to.be.equal("peerA")
+    expect(scoreSpy.getCall(0).returnValue).to.be.equal(0)
+    expect(scoreSpy.getCall(1)).to.be.undefined
+    // 2nd time, use a cached score since it's white listed
+    gossipsub._acceptFrom("peerA")
+    expect(scoreSpy.getCall(1)).to.be.undefined
+  })
+
+  it('should recompute score after 1s', () => {
+    // by default the score is 0
+    gossipsub._acceptFrom("peerA")
+    // 1st time, we have to compute score
+    expect(scoreSpy.getCall(0).args[0]).to.be.equal("peerA")
+    expect(scoreSpy.getCall(1)).to.be.undefined
+    gossipsub._acceptFrom("peerA")
+    // score is cached
+    expect(scoreSpy.getCall(1)).to.be.undefined
+
+    // after 1s
+    sandbox.clock.tick(1001)
+
+    gossipsub._acceptFrom("peerA")
+    expect(scoreSpy.getCall(1).args[0]).to.be.equal("peerA")
+    expect(scoreSpy.getCall(2)).to.be.undefined
+  })
+
+  it('should recompute score after max messages accepted', () => {
+    // by default the score is 0
+    gossipsub._acceptFrom("peerA")
+    // 1st time, we have to compute score
+    expect(scoreSpy.getCall(0).args[0]).to.be.equal("peerA")
+    expect(scoreSpy.getCall(1)).to.be.undefined
+
+    for (let i = 0; i < 128; i++) {
+      gossipsub._acceptFrom("peerA")
+    }
+    expect(scoreSpy.getCall(1)).to.be.undefined
+
+    // max messages reached
+    gossipsub._acceptFrom("peerA")
+    expect(scoreSpy.getCall(1).args[0]).to.be.equal("peerA")
+    expect(scoreSpy.getCall(2)).to.be.undefined
+  })
+
+  // TODO: run this in a unit test setup
```

Comment:
there's an issue that sinon causes the test to never finish in `firefox` environment. I think current test set up is like an e2e test where we'll run it in different OSs and browsers, and sinon is not expected to run on it, we need a separate `test:unit` where we can stub any functions